### DISCUSSION TITLE
What is heatmap and GPU Mode meaning in keyboard backlight setting?

### DISCUSSION THREAD

**Discussion by MC-dusk** [seerge/g-helper#1268] [Q&A] [answered]
I'm using TUF Gaming A15, and switch to heatmap and GPU Mode in keyboard backlight setting, but nothing special happens, and it just changes my light color.
Are those modes available in my PC?

**Comment by seerge** [comment]
It colors your keyboard proportionately to your CPU temp

**Reply by MC-dusk** [reply]
Do you mean heatmap mode will change my keyboard light color which is related to the CPU temp?
What about GPU Mode, is it corresponding to GPU temp?

**Reply by seerge** [reply, marked as answer]
Yes, Heatmap will change from Green to Yellow and Red based on temp
GPU mode just changes color based on current GPU https://github.com/seerge/g-helper/issues/1068

**Reply by MC-dusk** [reply]
Thank you, the CPU mode color seems changed obviously when I'm running games (90℃), but GPU mode color is always green, even it's 70℃, I'll keep observing it.
Maybe it's just beacuse the GPU temp is not too high. For the most of the time, the temp interval (about 45~65℃) won't change the color.

**Reply by seerge** [reply]
GPU mode changes color based on your active "GPU mode" in the app. If you are in Eco it's green, in standard - yellow, ultimate -red.

**Reply by MC-dusk** [reply]
Sorry, I misunderstand, then that's all right.

**Comment by MC-dusk** [comment]
I received a weird mail notification from _ameliajhons_ said:

> A heatmap in keyboard backlight settings visually displays frequently used keys. "GPU Mode" synchronizes lighting effects with GPU activity for dynamic illumination during gaming or visual tasks

Though it's not true in this scene, I have to admit that I also think so before posting this discussion.

**Comment by Nahiantheballer** [comment]
This helps.
